contrib/cmake/build3rdParty/build3rdParty.cpp

```diff
@@ -0,0 +1,273 @@
+                }
+                cout << "Copying triplet from SDK: " << triplet << endl;
+                fs::copy(sdkRootPath / "contrib" / "cmake" / "vcpkg_extra_triplets" / (triplet+".cmake"), vcpkgDir / "triplets" / (triplet + ".cmake"));
+            }
+            else if (!fs::exists(vcpkgDir / "triplets" / (triplet + ".cmake")))
+            {
+                cout << "triplet not found in the SDK or in vcpkg: " << triplet << endl;
+                exit(1);
+            }
+
+            for (auto portPair : ports)
+            {
+                const string& portname = portPair.first;
+                const string& portversion = portPair.second;
+
+                if (fs::is_directory(fs::path("ports") / portname))
+                {
+                    cout << "Removing " << (vcpkgDir / "ports" / portname).u8string() << endl;
+                    fs::remove_all(vcpkgDir / "ports" / portname);
+                }
+                if (portversion.size() == 40)
+                {
+                    fs::current_path(cloneDir);
+                    execute("git checkout --quiet " + portversion);
+                    cout << "Copying port for " << portname << " from vcpkg commit " << portversion << endl;
+                    fs::copy(cloneDir / "ports" / portname, vcpkgDir / "ports" / portname, fs::copy_options::recursive);
+                    fs::current_path(vcpkgDir);
+                }
+                else
+                {
+                    cout << "Copying port for " << portname << " from SDK customized port " << portversion << endl;
+                    fs::copy(sdkRootPath / "contrib" / "cmake" / "vcpkg_extra_ports" / portname / portversion, vcpkgDir / "ports" / portname, fs::copy_options::recursive);
+                }
+            }
+
+            if (removeUnusedPorts)
+            {
+                for (auto dir = fs::directory_iterator(vcpkgDir / "ports"); dir != fs::directory_iterator(); ++dir)
+                {
+                    if (ports.find(dir->path().filename().u8string()) == ports.end())
+                    {
+                        fs::remove_all(dir->path());
+                    }
+                }
+            }
+
+        }
+        else if (build)
+        {
+
+            if (!fs::is_directory("vcpkg"))
+            {
+                cout << "This command should be run from just outside 'vcpkg' folder - maybe it is not set up?" << endl;
+                return 1;
+            }
+            else
+            {
+                fs::current_path("vcpkg");
+            }
+
+
+            for (auto portPair : ports)
+            {
+                #ifdef WIN32
+                    execute("vcpkg install --triplet " + triplet + " " + portPair.first);
+                #else
+                    execute("./vcpkg install --triplet " + triplet + " " + portPair.first);
+                #endif
+            }
+        }
+
+
+    }
+
+    return 0;
+}
+catch (exception& e)
```

Comment:
Might as well catch by `const exception&` since we're not mutating it?

Reply:
Personally I find the extra syntax for 'const' unnecessary - you can see at a glance that it's not changed in such a small piece of code.   And if we did change it, well, there's no other piece of code that we would be breaking by doing that.   If it's a longer function where one might legitimately worry about something being modified when it shouldn't be then of course const is useful in the declaration, but I don't think we need to bulk up the code by adding it in every place that it could fit.